src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return (
+    sourceString
+      // Breaks sourceString by semicolon into separate stings
+      .split(';')
+
+      // Breaks line into property and value and removes extra space charachters
+      .map((el) => el.split(':').map((x) => x.trim()))
+
+      // Leaves only valid elements
+      .filter((el) => el.length === 2)
+
+      // Generate object from the
+      .reduce((acc, [item, value]) => ({ ...acc, [item]: value }), {})
+  );
```

Comment:
The use of the spread operator inside the 'reduce' callback creates a new object on every iteration which is unnecessary and can lead to performance issues. Consider using the accumulator directly without spreading into a new object.